### PR TITLE
Add rate-limiting

### PR DESCRIPTION
**Pull request by Hermsi1337**
I tried to add rate-limiting capability.

Since I'm not very familiar with GO-Templating I'm not sure if I did it correct.
I simply tried to adapt the scheme of the template. ^_^

Feedback is appreciated.

PS: See #59 

Regards
Hermsi

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,9 @@ Traefik labels, has to be created in your service or externalService, in order t
 - traefik.path.strip = < path >		       	# Path strip rule. Multiple values separated by ","
 - traefik.path.prefix = < path >	       	# Path prefix rule. Multiple values separated by ","
 - traefik.path.prefix.strip = < path >	   	# Path prefix strip rule. Multiple values separated by ","
+- traefik.ratelimit.period = < ns >         # Replace n with desired amount of seconds in which traefik is checking the limits "average" and "burst", e.g. 10s
+- traefik.ratelimit.average = < n >         # Change to desired average allowed requests by client ip, e.g. 100
+- traefik.ratelimit.burst = < n >           # State what limit the client ip is allowed to burst up to respectively, e.g. 200
```
Comment (rawmind0) [inline README.md]: What do you think if you add a label to enable ratelimit??
```
+- traefik.ratelimit.enable = < true | false >           # Enable/disable rate limit.
```

File: root/opt/tools/rancher-template/tmpl/traefik.tmpl [hunk 2]
```diff
@@ -154,6 +154,28 @@
...
+        {{- if gt $traefik_ratelimit_average 0}}
+        average = $traefik_ratelimit_average
+        {{- else}}
+        average = 100
+        {{- if gt $traefik_ratelimit_burst 0}}
+        burst = $traefik_ratelimit_burst
+        {{- else}}
+        burst = 200
+        {{- end}}
+    {{- end -}}
```
Comment (rawmind0) [inline root/opt/tools/rancher-template/tmpl/traefik.tmpl]: Set rate limit config based on `traefik.ratelimit.enable` label. 
Also corrected indent, other syntax errors and code refactored....

```
+      {{- $traefik_ratelimit_enable := (index $service.Labels "traefik.ratelimit.enable") -}}
+      {{- if eq $traefik_ratelimit_enable "true" -}}
+        {{- $traefik_ratelimit_period := (index $service.Labels "traefik.ratelimit.period") -}}
+        {{- $traefik_ratelimit_average := (index $service.Labels "traefik.ratelimit.average") -}}
+        {{- $traefik_ratelimit_burst := (index $service.Labels "traefik.ratelimit.burst") }}
+    [frontends.{{$service.Name}}__{{$stack.Name}}.ratelimit]
+      extractorfunc = "client.ip"
+      [frontends.{{$service.Name}}__{{$stack.Name}}.ratelimit.rateset.rateset1]
+        period = {{if gt (len $traefik_ratelimit_period) 0 -}}"{{$traefik_ratelimit_period}}s"
+                 {{- else -}} "10s"{{- end }}
+        average = {{if gt (len $traefik_ratelimit_average) 0 -}}{{$traefik_ratelimit_average}}
+                  {{- else -}}100{{- end}}
+        burst = {{if gt (len $traefik_ratelimit_burst) 0}}{{$traefik_ratelimit_burst}}
+                {{- else -}}200{{- end}}
+      {{- end -}}
```

File: README.md [hunk 3]
```diff
@@ -45,6 +45,9 @@ Traefik labels, has to be created in your service or externalService, in order t
 - traefik.path.strip = < path >		       	# Path strip rule. Multiple values separated by ","
 - traefik.path.prefix = < path >	       	# Path prefix rule. Multiple values separated by ","
 - traefik.path.prefix.strip = < path >	   	# Path prefix strip rule. Multiple values separated by ","
+- traefik.ratelimit.period = < ns >         # Replace n with desired amount of seconds in which traefik is checking the limits "average" and "burst", e.g. 10s
```
Comment (rawmind0) [inline README.md]: Don't add the `s` in the label, just the number...
```
-- traefik.ratelimit.period = < ns >         # Replace n with desired amount of seconds in which traefik is checking the limits "average" and "burst", e.g. 10s
+- traefik.ratelimit.period = < n >         # Replace n with desired amount of seconds in which traefik is checking the limits "average" and "burst", e.g. 10s
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]

**Comment by rawmind0** [comment]
Hi @Hermsi1337 , thanks for the PR but the template is not working....

I've refactored and tested the code, please take a look to the changes requested....

**Comment by Hermsi1337** [comment]
Many thanks for your fixes and sorry for my syntax-failure.

I hope that everythings fine now! (=

**Review by rawmind0 — APPROVED** [review]

**Comment by rawmind0** [comment]
LGTM...

**Comment by rawmind0** [comment]
Released rawmind/rancher-traefik:1.5.0-0

**Pull request merged**